components/scream/scripts/test_all_scream.py

```diff
@@ -45,9 +47,32 @@ def __init__(self, cxx, kokkos, submit, parallel, fast_fail, baseline_ref, basel
                        "Requested test '{}' is not supported by test-all-scream, please choose from: {}".\
                            format(t, ", ".join(self._test_full_names.keys())))
 
-        if not self._baseline_dir == "NONE":
-            print ("Ignoring baseline ref {}, and using baselines in directory {} instead".format(self._baseline_ref, self._baseline_dir))
-            print ("NOTE: baselines for each build type BT must be in '{}/BT/data'. We don't check this, but there will be errors if the baselines are not found.".format(self._baseline_dir))
+        expect(self._src_dir.endswith("components/scream"), "Run from $scream_repo/components/scream")
+        if not self._kokkos:
+            expect(self._machine, "If no kokkos provided, must provide machine name for internal kokkos build")
+        if self._submit:
+            expect(self._machine, "If dashboard submit request, must provide machine name")
+
+        # Compute baseline info
+        import pdb
+        pdb.set_trace()
+        expect(not (self._baseline_ref and self._baseline_dir),
+               "Makes no sense to specify a baseline generation commit if using pre-existing baselines ")
+        if self._baseline_dir is None:
+            if self._baseline_ref is None:
+                # Compute baseline ref
+                if self._keep_tree:
+                    self._baseline_ref = "HEAD"
```

Comment:
It makes no sense to use the `--keep-tree` flag together with a baseline ref different from HEAD. In this series of if's, we process the `--keep-tree` flag only if there is no baseline selected. Could we check that first? I want to avoid some errors due to copy+paste, where `--keep-tree` is ignored because a baseline ref was specified, and the user is confused.

Reply:
There's a check later in this function that, if keep_tree is set, it's OK for the repo to be dirty. Keep-tree doesn't really need to do anything else.